pkg/apis/resource/validation/validation.go

```diff
@@ -912,3 +917,37 @@ func validateNetworkDeviceData(networkDeviceData *resource.NetworkDeviceData, fl
 		fldPath.Child("ips"))...)
 	return allErrs
 }
+
+// AuthorizedForAdmin checks if the request is authorized to get admin access to devices
+// based on namespace label
+func AuthorizedForAdmin(ctx context.Context, deviceRequests []resource.DeviceRequest, namespaceName string, nsClient v1.NamespaceInterface) field.ErrorList {
```

Comment:
this validation should live in pkg/registry/resource/resourceclaim, not in API validation (we don't pass API clients down into validation and do object lookups there)

Reply:
Ok will create a common file and move this there so resourceclaim and resourceclaimtemplate can share